lsm-perf.py

```diff
@@ -0,0 +1,180 @@
+
+    def scp_to(self, src_local, dst_remote):
+        """Send a file from the host to the VM"""
+        assert self.ssh is not None
+        fro = local.path(src_local)
+        to = self.ssh.path(dst_remote)
+        plumbum.path.utils.copy(fro, to)
+
+
+def construct_qemu_args(kernel_path, img_path):
+    """Qemu arguments similar to what `vm start` produces"""
+    return [
+        '-nographic',
+        '-s',
+        '-machine', 'accel=kvm',
+        '-cpu', 'host',
+        '-device', 'e1000,netdev=net0',
+        '-netdev', 'user,id=net0,hostfwd=tcp::5555-:22',
+        '-append', 'console=ttyS0,115200 root=/dev/sda rw nokaslr',
+        '-smp', '4',
+        '-m', '4G',
+        '-drive', 'if=none,id=hd,file=%s,format=raw' % img_path,
+        '-device', 'virtio-scsi-pci,id=scsi',
+        '-device', 'scsi-hd,drive=hd',
+        '-device', 'virtio-rng-pci,max-bytes=1024,period=1000',
+        '-qmp', 'tcp:localhost:4444,server,nowait',
+        '-serial', 'mon:stdio',
+        '-kernel', '%s' % kernel_path,
+        '-name', 'lsm_perf_vm,debug-threads=on'
+    ]
+
+
+def print_eta(kernel_name, info=""):
+    """Updates the status of the evaluation of a kernel"""
+    sys.stdout.write('\r\tEvaluating %s: %s' % (kernel_name, info) + ' ' * 20)
+    sys.stdout.flush()
+
+
+def init_output_file(file):
+    """Writes the header in the result file"""
+    columns = (['kernel path', 'round'] +
+               ['run %d' % i for i in range(REPETITIONS_PER_ROUND)])
+    file.write(','.join(columns) + '\n')
+
+
+def write_results_to_file(file, kernel_path, round, results):
+    """Writes the results of the evaluation of a kernel to the file"""
+    row = [kernel_path, round] + results
+    file.write(','.join([str(x) for x in row]) + '\n')
+    file.flush()
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description=('Compares the performances of several kernels'
+                     ' on the same workload.'))
+    parser.add_argument(
+        '-i', '--image', type=argparse.FileType('r'), required=True,
+        help='Path of the disk image to boot the kernels from.')
+    parser.add_argument(
+        '-k', '--kernels', type=argparse.FileType('r'), required=True,
+        help='Path of all the kernels to evaluate.', nargs='+')
+    parser.add_argument(
+        '-w', '--workload', type=argparse.FileType('r'), required=True,
+        help=('Path of the workload program to run to evaluate the kernels. '
+              'This should take no argument, and simply output an integer '
+              'to stdout (the time measurement)'))
+    parser.add_argument(
+        '-key', type=argparse.FileType('r'), default='~/.ssh/id_rsa',
+        help=('Path of the RSA key to connect to the VM. '
+              'It must be in the list of authorized keys in the image.'))
+    parser.add_argument(
+        '-o', '--out', type=argparse.FileType('w'), default='lsm-perf.csv',
+        help='Path of the output file.')
```

Comment:
Nice - this is good and clear and it's nice to see the cmdline args decoupled from the `VM` class via `main`.